src/Sulu/Bundle/AdminBundle/Resources/js/views/List/List.js

```diff
@@ -168,17 +173,26 @@ class List extends React.Component<ViewProps> {
         this.listStore.destroy();
     }
 
-    addItem = (rowId: string | number) => {
-        const {router} = this.props;
-        const {
-            route: {
-                options: {
-                    addRoute,
-                },
-            },
-        } = router;
+    handleItemAdd = (parentId: string | number) => {
+        const {onItemAdd, router} = this.props;
+
+        if (onItemAdd) {
+            onItemAdd(parentId);
+            return;
+        }
+
+        router.navigate(router.route.options.addRoute, {locale: this.locale.get(), parentId: parentId});
```

Comment:
I would still prefer to destruct these values, because it is easier to add more stuff later on then. And `parentId: parentId` is redundant and can be replaced by just `parentId`.